src/python/tests/manifest/test_manifest_stats.py

```diff
@@ -0,0 +1,306 @@
+            },
+            id="1 gene, 1 translation",
+        ),
+        param(
+            "get_gff3/te_gene.gff",
+            {
+                "gff3_transposable_elements": {"LOREMIPSUM1": 1000},
+            },
+            id="1 TE",
+        ),
+        param(
+            "get_gff3/pseudogene_cds.gff",
+            {
+                "gff3_genes": {"LOREMIPSUM1": 1000},
+                "gff3_translations": {},
+                "gff3_all_translations": {"LOREMIPSUM1_t1_cds": 266},
+            },
+            id="1 pseudogene with CDS",
+        ),
+    ],
+)
+def test_get_gff3(
+    tmp_path: Path,
+    data_dir: Path,
+    gff3_path: str | None,
+    expected_data: dict[str, int],
+) -> None:
+    """Tests `ManifestStats.get_gff3()`."""
+    if gff3_path:
+        copy(data_dir / gff3_path, tmp_path / "test.gff3")
+    manifest = Manifest(tmp_path)
+    manifest.create()
+    stats = ManifestStats(manifest.dir / "manifest.json")
+    stats.get_gff3()
+    expected_lengths = {**stats.lengths, **expected_data}
+    assert stats.lengths == expected_lengths
+
+
+def test_get_genome(tmp_path: Path) -> None:
+    """Tests `ManifestStats.get_genome()`."""
+    genome_data = {"LOREM": "1"}
+    genome_path = tmp_path / "genome.json"
+    print_json(genome_path, genome_data)
+    manifest = Manifest(tmp_path)
+    manifest.create()
+    stats = ManifestStats(manifest.path)
+    stats.get_genome()
+    assert stats.genome == genome_data
+
+
+def test_prepare_integrity_data(tmp_path: Path) -> None:
+    """Tests `ManifestStats.prepare_integrity_data()`."""
+    manifest_path = tmp_path / "manifest.json"
+    print_json(manifest_path, {})
+    stats = ManifestStats(manifest_path)
+    stats.prepare_integrity_data()
+    assert stats.lengths == {
+        "agp": {},
+        "ann_genes": {},
+        "ann_translations": {},
+        "ann_transposable_elements": {},
+        "annotations": {},
+        "dna_sequences": {},
+        "gff3_all_translations": {},
+        "gff3_genes": {},
+        "gff3_seq_regions": {},
+        "gff3_translations": {},
+        "gff3_transposable_elements": {},
+        "peptide_sequences": {},
+        "seq_region_levels": {},
+        "seq_regions": {},
+    }
+
+
+def test_has_lengths(manifest_path: Path) -> None:
+    """Tests `ManifestStats.has_lengths()`."""
+    stats = ManifestStats(manifest_path)
+    with raises(KeyError):
+        stats.has_lengths("foobar")
+    assert not stats.has_lengths("ann_genes")
+    stats.get_functional_annotation()
+    assert stats.has_lengths("ann_genes")
+
+
+def test_get_lengths(manifest_path: Path) -> None:
+    """Tests `ManifestStats.get_lengths()`."""
+    stats = ManifestStats(manifest_path)
+    with raises(KeyError):
+        stats.get_lengths("foobar")
+    assert not stats.has_lengths("ann_genes")
+    stats.get_functional_annotation()
+    circular = stats.get_lengths("ann_genes")
+    assert circular == {"ECC02_000372": 1}
```

Comment:
I know it is tedious, but these should be 2 different test parametrisations.

Furthermore, I believe the line
```
assert not stats.has_lengths("ann_genes")
```
should not be here.